dependencies/src/main/java/com/google/cloud/tools/opensource/lts/LtsRuntimeCompatibilityTest.java

```diff
@@ -0,0 +1,247 @@
+      }
+
+      // Build the project
+
+      Path shellScript = testRoot.resolve("test_" + i++ + ".sh");
+      String shellScriptLocation = shellScript.toAbsolutePath().toString();
+      com.google.common.io.Files.asCharSink(shellScript.toFile(), Charsets.UTF_8).write(commands);
+
+      Process buildProcess = Runtime.getRuntime()
+          .exec(String.format("/bin/bash %s", shellScriptLocation),
+              null, projectDirectory.toFile());
+
+      com.google.common.io.Files.asCharSink(projectDirectory.resolve("stdout.log").toFile(),
+          Charsets.UTF_8, FileWriteMode.APPEND).writeFrom(new InputStreamReader(buildProcess.getInputStream()));
+      com.google.common.io.Files.asCharSink(projectDirectory.resolve("stderr.log").toFile(),
+          Charsets.UTF_8, FileWriteMode.APPEND).writeFrom(new InputStreamReader(buildProcess.getErrorStream()));
+
+      int buildStatusCode = buildProcess.waitFor();
+
+      if (buildStatusCode != 0) {
+        System.out
+            .println("Failed to build " + url + ". Exiting. Check the logs in " + projectDirectory);
+        break;
+      } else {
+        System.out.println(name + " passed!");
+      }
+    }
+  }
+
+  static void modifyGradleFiles(Path projectRoot, Bom bom) throws IOException {
+    Iterable<Path> paths = MoreFiles.fileTraverser().breadthFirst(projectRoot);
+
+    for (Path path : paths) {
+      if (!path.getFileName().endsWith("build.gradle")) {
+        continue;
+      }
+
+      modifyGradleFile(path, bom);
+    }
+  }
+
+  static void modifyGradleFile(Path gradleFile, Bom bom) throws IOException {
+
+    List<String> lines = com.google.common.io.Files.readLines(gradleFile.toFile(), Charsets.UTF_8);
+
+    String bomCoordinates= bom.getCoordinates();
+    
+
+    
+    List<String> replacedLines = lines.stream().map(line -> line.replaceAll("^dependencies \\{",
+        "dependencies {\n    testRuntime enforcedPlatform('"+bomCoordinates+"')")).collect(Collectors.toList());
+
+    com.google.common.io.Files.asCharSink(gradleFile.toFile(),
+        Charsets.UTF_8).write(Joiner.on("\n").join(replacedLines));
+  }
+
+  static void modifyPomFiles(Path projectRoot, Bom bom) throws IOException, ParsingException {
+    Iterable<Path> paths = MoreFiles.fileTraverser().breadthFirst(projectRoot);
+
+    for (Path path : paths) {
+      if (!path.getFileName().endsWith("pom.xml")) {
+        continue;
+      }
+
+      modifyPomFile(path, bom);
+    }
+  }
+
+  static ImmutableMap<String, String> versionlessCoordinatesToVersion(Bom bom) {
+    ImmutableMap.Builder<String, String> map = ImmutableMap.builder();
+
+    for (Artifact managedDependency : bom.getManagedDependencies()) {
+      String artifactId = managedDependency.getArtifactId();
+      if ("guava".equals(artifactId)) {
+        // Guava JRE does not work with Yoshi's shared config.
+        // https://github.com/GoogleCloudPlatform/cloud-opensource-java/issues/1974#issuecomment-799862063
+        continue;
```

Comment:
This is an unexpected blocker when building the project with Guava's JRE flavor.